fem/transfer.hpp

```diff
@@ -305,13 +305,130 @@ class L2ProjectionGridTransfer : public GridTransfer
       virtual void ProlongateTranspose(const Vector& x, Vector& y) const;
       virtual void SetRelTol(double p_rtol_);
       virtual void SetAbsTol(double p_atol_);
+   protected:
+      /// Sets default shared options for the CGSolver
+      void InitializeCGSolver();
+      /// Computes on-rank R and M_LH matrices
+      std::pair<SparseMatrix*, SparseMatrix*> ComputeSparseRAndM_LH();
+      void GetDofsByVDim(const FiniteElementSpace& fes,
+                         int vdim, const Vector& x, Vector& x_vdim) const;
+      void SetFromDofsByVDim(const FiniteElementSpace& fes,
+                             int vdim, const Vector& y_vdim, Vector& y) const;
+      /// Abstract method to return the inverse of an on-rank lumped mass matrix
+      virtual void LumpedMassInverse(Vector& ML_inv) const = 0;
+      /// Sets <tt>x_vdim_true</tt>, true dof values on a chosen <tt>vdim</tt>
+      /// given vector dof values <tt>x</tt> on the coarse (higher-order) mesh.
+      virtual void GetHOTDofsByVDim(int vdim, const Vector& x,
+                                    Vector& x_vdim_true) const = 0;
+      /// Sets <tt>x_vdim_true</tt>, true dof values on a chosen <tt>vdim</tt>
+      /// given vector dof values <tt>x</tt> on the refined (low-order) mesh.
+      virtual void GetLORTDofsByVDim(int vdim, const Vector& x,
+                                     Vector& x_vdim_true) const = 0;
+      /// Sets the true dof values on a chosen <tt>vdim</tt> from
+      /// <tt>y_vdim_true</tt> onto <tt>y</tt>, vector dof values on the coarse
+      /// (higher-order) mesh.
+      virtual void SetHOFromTDofsByVDim(int vdim, const Vector& y_vdim_true,
+                                        Vector& y) const = 0;
+      /// Sets the true dof values on a chosen <tt>vdim</tt> from
+      /// <tt>y_vdim_true</tt> onto <tt>y</tt>, vector dof values on the refined
+      /// (low-order) mesh.
+      virtual void SetLORFromTDofsByVDim(int vdim, const Vector& y_vdim_true,
+                                         Vector& y) const = 0;
    private:
       /// Computes sparsity pattern and initializes R matrix. Based on
-      /// BilinearForm::AllocMat() except maps between HO elements and LOR
-      /// elements.
-      void AllocR();
+      /// BilinearForm::AllocMat() except maps between coarse HO elements and
+      /// refined LOR elements.
+      SparseMatrix* AllocR();
    };
 
+   /** Implements class for projection operator between a H1 high-order finite
+       element space on a coarse mesh, and a H1 low-order finite element space
+       on a refined mesh (LOR) in serial. */
+   class SerialL2ProjectionH1Space : public L2ProjectionH1Space
```

Comment:
The typical MFEM pattern is that the base class (`L2ProjectionH1Space` in this case) implements the serial functionality, and the parallel class inherits from the serial class. Following that pattern here as well could make the code easier to understand for others familiar with the usual MFEM way.